src/timezones/Olson.jl

```diff
@@ -332,7 +341,10 @@ function resolve!(zone_name::AbstractString, zoneset::ZoneDict, ruleset::RuleDic
             end
         else
             if !haskey(ordered, rule_name)
-                ordered[rule_name] = order_rules(ruleset[rule_name]; max_year=max_year)
+                dates, ordered_rules, was_truncated = order_rules(ruleset[rule_name]; max_year=max_year)
+
+                ordered[rule_name] = dates, ordered_rules
+                truncated |= was_truncated
```

Comment:
The truncated information will be lost on future passes where `rule_name` is a key within `ordered`.